boards/nuvoton/numaker_m2l31ki/numaker_m2l31ki.dts

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2024 Nuvoton Technology Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/dts-v1/;
+
+#include <nuvoton/m2l31kid.dtsi>
+#include "numaker_m2l31ki-pinctrl.dtsi"
+#include <zephyr/dt-bindings/input/input-event-codes.h>
+
+/ {
+	model = "Nuvoton NuMaker M2L31KI board";
+	compatible = "nuvoton,numaker-m2l31ki";
+
+	aliases {
+		led0 = &red_led;
+	};
+
+
+	chosen {
+		zephyr,console = &uart0;
+		zephyr,shell-uart = &uart0;
+		zephyr,sram = &sram0;
+		zephyr,flash = &flash0;
+		zephyr,code-partition = &slot0_partition;
+	};
+
+
+	leds {
+		compatible = "gpio-leds";
+		red_led: led_0 {
+			gpios = <&gpioc 14 GPIO_ACTIVE_LOW>;
+			label = "User LD1";
+		};
+	};
+};
+
+&scc {
+	/* For USB 1.1 Host/Device/OTG, configure to 192MHz, which can generate necessary 48MHz. */
+	/* For USB 2.0 Host/Device/OTG or no USB application, comment out to use default. */
+	core-clock = <DT_FREQ_M(192)>;
+};
+
+&gpioc {
+	status = "okay";
+};
+
+&flash0 {
+	partitions {
+		compatible = "fixed-partitions";
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		slot0_partition: partition@0 {
```

Comment:
How are these slots used if there is no bootloader?

Reply:
These proposed slots should be used by bootloader.

In \tests\drivers\flash\common\src, it will utilize the
    `#define TEST_AREA	storage_partition.`

Reply:
Application has update client capacity and could download new firmware & write it into slot1 in runtime.
As while bootloader detect new firmware in slot1, it could clone into slot0 and set as active image.

Reply:
No, it's not the mask ROM.
Slot0 & Slot1 belong to APROM.
The chip has one LDROM ( loader-rom), user could program the loader (i.e small firmware).

Reply:
My question is do these work out of the box or does someone need to build and load their own bootloader e.g. mcuboot? If this works out of the box then this can be marked as resolved, if they need e.g. mcuboot or another bootloader which they need to flash themselves then there should be a partition here for that bootloader

Reply:
It need one bootloader, so I add one more partition for loader in numaker_m2l31ki.dts .
Update commit  f93489d3cca67.

Reply:
Is slot2 the bootloader? If so, how does that code partition run first? If you are using mcuboot could you copy the naming from here: https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/nordic/nrf52840dk/nrf52840dk_nrf52840.dts#L277

Reply:
Yes, it could be mucboot or another simple bootloader.
Thanks and it's OK to use the same naming.